lib/services/devices/deviceService.js

```diff
@@ -292,13 +351,13 @@ function unregisterDevice(id, callback) {
 
     logger.debug('Removing device register in Device Service');
 
-    config.getRegistry().get(id, function(error, device) {
+    config.getRegistry().get(id, service, subservice, function(error, device) {
         async.waterfall([
             async.apply(removeAllSubscriptions, device),
             processUnsubscribes,
             async.apply(registrationUtils.sendRegistrations, true, device),
             processContextUnregister,
-            async.apply(config.getRegistry().remove, id)
+            async.apply(config.getRegistry().remove, id, service, subservice)
```

Comment:
There are some occurrences of `async.apply` which could directly be substituted by `apply`.